meta.jq

```diff
@@ -53,10 +53,24 @@ def pull_command:
 	end
 ;
 # input: "build" object (with "buildId" top level key)
+# output: string "giturl" ("https://github.com/docker-library/golang#commit:directory), used for "docker build giturl"
+def git_build_url:
+	.source.entry.GitRepo as $gitRepo
+	| if ($gitRepo | test("^https?://github.com/"))
+		and ($gitRepo | endswith(".git") | not) then
+			# without ".git" in the url "docker build url" fails and tries to build the html repo page as a Dockerfile
+			# https://github.com/moby/moby/blob/4f9c865eddc09da0c0cb9fe08c76b81b804093f5/builder/remotecontext/urlutil/urlutil.go#L15
+			$gitRepo + ".git"
+	else
+		$gitRepo
+	end
+	+ "#" + .source.entry.GitCommit + ":" + .source.entry.Directory
```

Comment:
Even better we can simplify the whole block with less `()` by putting some on the outside and ditching the variable:

```suggestion
	.source.entry
	| (
		.GitRepo
		| if (endswith(".git") | not) then
			if test("^https?://github.com/") then
				# without ".git" in the url "docker buildx build url" fails and tries to build the html repo page as a Dockerfile
				# https://github.com/moby/buildkit/blob/0e1e36ba9eb8142968b2c5cfa2f12549bf9246d9/util/gitutil/git_ref.go#L81-L87
				. + ".git"
			else
				error("\(.) does not end in '.git' so build will fail to recognize it as a Git URL")
			end
		else . end
	) + "#" + .GitCommit + ":" + .Directory
```

(this also adds an error for anything we know `docker buildx build` will fail inscrutably with)

```console
$ jq -nc '"https://github.com/foo/bar" | if (endswith(".git") | not) then
			if test("^https?://github.com/") then
				# without ".git" in the url "docker buildx build url" fails and tries to build the html repo page as a Dockerfile
				# https://github.com/moby/buildkit/blob/0e1e36ba9eb8142968b2c5cfa2f12549bf9246d9/util/gitutil/git_ref.go#L81-L87
				. + ".git"
			else
				error("\(.) does not end in '.git' so build will fail to recognize it as a Git URL")
			end
		else . end'
"https://github.com/foo/bar.git"

$ jq -nc '"https://tianon.xyz/foo/bar" | if (endswith(".git") | not) then
			if test("^https?://github.com/") then
				# without ".git" in the url "docker buildx build url" fails and tries to build the html repo page as a Dockerfile
				# https://github.com/moby/buildkit/blob/0e1e36ba9eb8142968b2c5cfa2f12549bf9246d9/util/gitutil/git_ref.go#L81-L87
				. + ".git"
			else
				error("\(.) does not end in '.git' so build will fail to recognize it as a Git URL")
			end
		else . end'
jq: error (at <unknown>): https://tianon.xyz/foo/bar does not end in .git so build will fail to recognize it as a Git URL
```